access/ms-teams/README.md

```diff
@@ -0,0 +1,176 @@
+
+### Building from source
+
+To build the plugin from source you need [Go](https://go.dev/) and `make`.
+
+```bash
+$ git clone https://github.com/gravitational/teleport-plugins.git
+$ cd teleport-plugins/access/ms-teams
+$ make
+$ ./build/teleport-ms-teams start
+```
+
+
+
+## Teleport User and Role
+
+Using Web UI or `tctl` CLI utility, create the role `access-ms-teams` and the user `access-ms-teams` belonging to the role `access-ms-teams`. You may use the following YAML declarations.
+
+### Role
+
+```yaml
+kind: role
+metadata:
+  name: access-ms-teams
+spec:
+  allow:
+    rules:
+      - resources: ['access_request']
+        verbs: ['list', 'read', 'update']
+version: v5
+```
+
+### User
+
+```yaml
+kind: user
+metadata:
+  name: access-ms-teams
+spec:
+  roles: ['access-ms-teams']
+version: v2
+```
+
+## Configure Microsoft Bot and App
+
+Please note that you must have the appropriate Azure AD adminstrator permissions and MS Teams organization admin role to follow these steps. Contact your IT department administrator for details.
+
+1. [Register new Azure bot](https://portal.azure.com/#create/Microsoft.AzureBot). Set "Single Tenant" type of app. Choose "Create new Microsoft App ID".
+
+  ![Create Microsoft Bot](images/app_create.png)
+
+2. Open the newly created bot -> Channels tab. Add Microsoft Teams channel.
+
+  ![Add Bot channel](images/add_channel.png)
+
+3. Open the associated Microsoft App (Configuration tab, find "Microsoft App ID", click "Manage" link next to it).
+
+  ![Manage app](images/manage_app.png)
+
+4. Go to "Certificates & secrets" section. Click "New client secret". Press "Add". Copy and save the generated password in a secure location.
+
+  ![Add password](images/add_password.png)
+
+5. Go to "API permissions" section. Click "Add a permission". Choose "Microsoft Graph". Choose "Application permissions". Add the following permissions:
+
+* `TeamsAppInstallation.ReadWriteSelfForUser.All`
+* `AppCatalog.Read.All`
+* `User.Read.All`
+
+6. Grant admin consent for the newly added permissions.
+
+7. Goto the bot Configuration tab (back to step #3). Find "Microsoft App ID", "App Tenant ID" and run the following command:
+
+```sh
+$ ./teleport-ms-teams configure ~/ms_teams --appID <Microsoft App ID value> --tenantID <App Tenant ID value> --appSecret <password value saved on step #4>
+```
+
+`~ms_teams` must be a path to non existent folder. It will be created. All required files will be generated.
+
+8. There will be `app.zip` in the target directory. This archive contains MS Teams app. If you want to change the application name, URLs and other settings, please update `manifest.json` and recreate `app.zip` from `color.png`, `outline.png` and `manifest.json`.
+
+9. Upload the app to MS Teams organization store. Open "Apps", then "Manage your apps", click "Upload an app", choose "Upload an app to your org's app catalog", choose `app.zip` 
```

Comment:
I think it's worth mentioning that Microsoft Teams Free does not support apps at all.

I created an app and was trying to install it to my Teams org.
There's not even an `Apps` menu when using the Free version

We can add a note at the beginning.

Reply:
Was not aware of that. I have free account and it seems working for me.

Reply:
I guess, you need kind of Azure subscription.

Reply:
Possibly :thinking: 
I created an account just now, not even associated with my work email
I'll try to get an azure subscription
![image](https://user-images.githubusercontent.com/689271/181570390-06aa7c3d-18a3-4dda-8afb-65442865c57e.png)

Reply:
I've added the notice, anyway.